src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 13;
+        result = 31 * result + (width == null ? 0 : width.hashCode());
+        result = 31 * result + (length == null ? 0 : length.hashCode());
+        result = 31 * result + (color == null ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (this.color == null && this.width == null && this.length == null) {
+            return true;
+        }
```

Comment:
are you sure that this check is needed?

Reply:
yes, I am, because when 3 fields equals null, I should return true and equal hashCode()
![image](https://user-images.githubusercontent.com/59320919/101469176-10ad7400-394d-11eb-9de7-0cd810062499.png)
![image](https://user-images.githubusercontent.com/59320919/101469250-2d49ac00-394d-11eb-84dd-11839e0a80af.png)
![image](https://user-images.githubusercontent.com/59320919/101469279-38044100-394d-11eb-922f-7c17bcdf47ab.png)